integration-tests/mustache/src/main/java/org/apache/camel/quarkus/component/mustache/it/MustacheRoute.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.quarkus.component.mustache.it;
+
+import io.quarkus.runtime.annotations.RegisterForReflection;
+import org.apache.camel.BindToRegistry;
+import org.apache.camel.builder.RouteBuilder;
+
+@RegisterForReflection(fields = true, methods = false)
+public class MustacheRoute extends RouteBuilder {
```

Comment:
As I get it, all route classes containing `@BindToRegistry` would need to register fields for reflection in order to work in native mode. This behavior is linked to [this code](https://github.com/apache/camel/blob/master/core/camel-base/src/main/java/org/apache/camel/impl/engine/DefaultCamelBeanPostProcessor.java#L212..L217). Would it make sense to open a ticket to improve this ? like finding a way to make @BindToRegistry registering classes for reflection or customizing bean post processing for camel-quarkus ?

Reply:
+1